x-pack/legacy/plugins/apm/public/components/app/Settings/CustomizeUI/CustomLink/CustomLinkFlyout/FiltersSection.tsx

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import {
+  EuiButtonEmpty,
+  EuiFieldText,
+  EuiFlexGroup,
+  EuiFlexItem,
+  EuiSelect,
+  EuiSpacer,
+  EuiText,
+  EuiTitle
+} from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { isEmpty } from 'lodash';
+import React, { useRef } from 'react';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { FilterOptionsType } from '../../../../../../../../../../plugins/apm/server/routes/settings/custom_link';
+import {
+  getSelectOptions,
+  DEFAULT_OPTION,
+  filterSelectOptions,
+  Filters
+} from './helper';
+
+export const FiltersSection = ({
+  filters,
+  onChangeFilters
+}: {
+  filters: Filters;
+  onChangeFilters: (filters: Filters) => void;
+}) => {
+  const filterValueRefs = useRef<HTMLInputElement[]>([]);
+
+  const onChangeFilter = (filter: Filters[0], idx: number) => {
+    if (filterValueRefs.current[idx]) {
+      filterValueRefs.current[idx].focus();
+    }
+    const copyOfFilters = [...filters];
+    copyOfFilters[idx] = filter;
+    onChangeFilters(copyOfFilters);
+  };
```

Comment:
`onChangeFilter` and `onRemoveFilter` is pretty hairy.  
imo it's not that important that important to have the ability to focus on the next input.

Reply:
@formgeist is it ok for you if we drop the focus for now?

Reply:
It's not a priority, just a nice UX and I guess also a11y-related thing. I'm fine with letting it not focus.